src/java.management/share/classes/javax/management/ObjectName.java

```diff
@@ -1222,7 +1222,7 @@ public static ObjectName getInstance(String domain,
     public static ObjectName getInstance(ObjectName name) {
         if (name.getClass().equals(ObjectName.class))
             return name;
-        return Util.newObjectName(name.getSerializedNameString());
+        return sun.management.Util.newObjectName(name.getSerializedNameString());
```

Comment:
The newObjectName methods are different in 2 Utils.
The share/classes/sun/management/Util.java is
```
  45   │     public static ObjectName newObjectName(String name) {
  46   │         try {
  47   │             return ObjectName.getInstance(name);
  48   │         } catch (MalformedObjectNameException e) {
  49   │             throw new IllegalArgumentException(e);
  50   │         }
  51   │     }
```
Isn't it infinite recursion?

Reply:
Right, they are different, but one uses "new", and one calls a newInstance method which uses "new", so they are really the same.  Daniel's note in the JBS bug hints that there may have been an idea to do some caching in getInstance, which has not as yet been implemented.